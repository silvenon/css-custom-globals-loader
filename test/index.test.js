const path = require('path')
const compile = require('./compile')
const cssGlobalsLoader = path.resolve(__dirname, '../')

it('emits a warning when not placed before css-loader', async () => {
  const entry = './fixtures/basic.css'
  const stats = await compile(entry, {
    cssRuleUse: ['css-loader', cssGlobalsLoader],
  })
  expect(stats.hasErrors()).toBe(true)
})

it('exposes CSS globals in the default export object', async () => {
  const entry = './fixtures/basic.css'
  const stats = await compile(entry, {
    cssRuleUse: [cssGlobalsLoader, 'css-loader'],
  })
  expect(stats.hasErrors()).toBe(false)
  const output = stats.toJson().modules.find(m => m.name.includes(entry)).source
  expect(output).toMatchSnapshot()
})

it('exposes CSS Modules in the same object as globals', async () => {
  const entry = './fixtures/modules.css'
  const stats = await compile(entry, {
    cssRuleUse: [cssGlobalsLoader, 'css-loader?modules'],
  })
  expect(stats.hasErrors()).toBe(false)
  const output = stats.toJson().modules.find(m => m.name.includes(entry)).source
  expect(output).toMatchSnapshot()
})
