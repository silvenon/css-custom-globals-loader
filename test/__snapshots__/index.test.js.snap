// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exposes CSS Modules in the same object as customs 1`] = `
"exports = module.exports = require(\\"../../node_modules/css-loader/lib/css-base.js\\")(false);
// imports


// module
exports.push([module.id, \\":root {\\\\n  --primary-color: lightblue;\\\\n}\\\\n\\\\n@custom-media --narrow-window (max-width: 30em);\\\\n\\\\n@custom-selector :--title h1;\\\\n\\\\n._3KySKA6rf07sl0ilq7bBBm {\\\\n  color: var(--primary-color);\\\\n}\\\\n\\", \\"\\"]);

// exports
exports.locals = {
	\\"text\\": \\"_3KySKA6rf07sl0ilq7bBBm\\"
};

exports.locals = Object.assign({}, exports.locals, {\\"customMedia\\":{\\"--narrow-window\\":\\"(max-width: 30em)\\"},\\"customProperties\\":{\\"--primary-color\\":\\"lightblue\\"},\\"customSelectors\\":{\\":--title\\":\\"h1\\"}});"
`;

exports[`exposes CSS customs in the default export object 1`] = `
"exports = module.exports = require(\\"../../node_modules/css-loader/lib/css-base.js\\")(false);
// imports


// module
exports.push([module.id, \\":root {\\\\n  --primary-color: lightblue;\\\\n}\\\\n\\\\n@custom-media --narrow-window (max-width: 30em);\\\\n\\\\n@custom-selector :--title h1;\\\\n\\", \\"\\"]);

// exports

exports.locals = Object.assign({}, exports.locals, {\\"customMedia\\":{\\"--narrow-window\\":\\"(max-width: 30em)\\"},\\"customProperties\\":{\\"--primary-color\\":\\"lightblue\\"},\\"customSelectors\\":{\\":--title\\":\\"h1\\"}});"
`;

exports[`supports files with external @imports 1`] = `
"exports = module.exports = require(\\"../../node_modules/css-loader/lib/css-base.js\\")(false);
// imports
exports.i(require(\\"-!../../node_modules/css-loader/index.js!normalize.css\\"), \\"\\");

// module
exports.push([module.id, \\"\\\\n\\", \\"\\"]);

// exports

exports.locals = Object.assign({}, exports.locals, {\\"customMedia\\":{},\\"customProperties\\":{},\\"customSelectors\\":{}});"
`;
